### PR TITLE
Add godoc comments for API condition types and remove unused conditions

### PR DESCRIPTION
**Pull request by pmorie**
Fill out godoc comments for API condition types and remove unused conditions from the API.

### CODE REVIEW

**Comment by MHBauer** [comment]
Going to edit the overall comments for Instance and Binding like Broker? 

Type and Status I agree need a comment reflecting the possibilities. Reason and Message feel repetitive as it's based on upstream. 

Can't deny wanting more comments.

Does code need to be regenerated?

File: pkg/apis/servicecatalog/v1alpha1/types.go [hunk 1]
```diff
@@ -204,31 +210,25 @@ type InstanceStatus struct {
...
+	Message string `json:"message"`
 }
 
 // InstanceConditionType represents a instance condition value
 type InstanceConditionType string
 
 const (
-	// InstanceConditionProvisioning represents that a given instance condition is in
-	// provisioning state
-	InstanceConditionProvisioning InstanceConditionType = "Provisioning"
```
Comment (vaikas-google) [inline pkg/apis/servicecatalog/v1alpha1/types.go]: Can you just leave these in? I'm working on 248 and was using these, so no point in removing and then re-adding?

Comment (pmorie) [inline pkg/apis/servicecatalog/v1alpha1/types.go]: I'm willing to leave it in for now, but we **should** bikeshed on it in your PR once you're done.  Does that work?

Comment (vaikas-google) [inline pkg/apis/servicecatalog/v1alpha1/types.go]: Yes :)

**Pull request merged**